pages/glossary/automatic-error-detection.md

```diff
@@ -0,0 +1,26 @@
+---
+title: Automatic detection of input errors
+key: automatic-error-detection
+---
+
+Automatic detection of input errors is an automatically carried procedure to verify that the provided input meets the requirements. 
```

Comment:
Another issue with this is that systems may not tell the user if an error has occurred. For example, I may build an e-shop where I let you optionally add your e-mail address. If you send me an invalid e-mail address, I might detect that but not tell you, and just to treat it as though you hadn't sent me anything. That automatic validation might even occur months later, as part of me sending my annual news letter.

I think we've got this a little backward. The problem shouldn't be about when an error occurs, but when an error is shown to the user.

Reply:
Agreed. What is relevant here is that there is an error message to present to the user, not that an error occurred. I'll try to rephrase it.